### ISSUE TITLE
Link credits of reprints automatically as first-level metadata with changes

### ISSUE THREAD

**Issue by LegalizeAdulthood** [GrandComicsDatabase/gcd-django#540]
Currently the credits are duplicated for a story that is reprinted in the issue where it is reprinted.  This can lead to divergence of credit information as the original is corrected and/or updated and this isn't tracked in the credits listed for the reprints.

This can't be done slavishly as sometimes things are changed during the reprint cycle: a cover might be modified, the colors might be redone, the colors might be dropped and the reprint is black and white only, etc.

However, for the most common case, everything is duplicated.

We have first-level metadata information for reprints as in "this story is reprinted from this other story in this other issue", but we don't have first-level metadata information for the credits of the reprinted story.

**Comment by jochengcd** [comment]
The most common case are actually foreign language reprints, where not everything is duplicated.

In any case, this ship has long sailed due to our our db-model. This could be solved in principle with an INDUCKS db-model, but it is unfortunately practically unfeasible to change for us.

The only thing we could provide at some point, is a credit check between reprint and original when editing the reprint, or vice versa, and support the updating of credits.

**Comment by LegalizeAdulthood** [comment]
Can you elaborate on what you mean by "long sailed due to our db-model"?  What is intrinsic to our data model that prohibits this layering and why can't we migrate the data model to something that supports it?

**Comment by ralfharing** [comment]
It would be a major re-architecting of the data model. It's always been that stories have exactly one parent issue, issues have exactly one parent series, and series have exactly one parent publisher. The model you're talking about (that INDUCKS has) is that a story can have many parent issues. That makes it possible to change the story record one time and have those changes be reflected in many issues, but it has its own challenges about how to then record differences.

It may be that the benefits outweigh the costs, but it'd be about as big of a change as the database would have ever undergone.

**Comment by LegalizeAdulthood** [comment]
I understand now, thanks for the explanation.  I plan to poke around the code and see if I can implement some of the issues I've opened and once I'm more familiar with the database model and code, I might be able to propose a path towards this.

I kinda found a way to do this manually by exporting the story credits as tsv and then importing them into the new issue that reprints them.  This is how I did this change: https://www.comics.org/changeset/6278592/compare/?collapse=1 and it saved me tons of time, although the exported tsvs didn't import cleanly without some hand editing (when field values contain embedded newlines).

However, this doesn't work so well for updating an existing credit set for an existing story in an existing reprint issue, only when creating new reprint issues.
